### PR TITLE
Switch from JSON configs to YAML configs

### LINKED ISSUE #32: Switch to YAML instead of JSON
We should be using YAML as it allows for comments and JSON does not.

### PR DESCRIPTION
**Pull request by treydock**
This will resolve #32 .

I kept the use of `json-schema` for validation instead of switching to a new library.

Added `scripts/json2yaml.rb` which was used to convert all the repo JSON files into YAML.

### CODE REVIEW

**Comment by ghoneycutt** [comment]
Not passing Travis yet, though the changes fundamentally look good to me.

**Comment by treydock** [comment]
Failures look like rubocop wants `YAML.safe_load` instead of `YAML.load` which is fine.  Also looks like should have used exception handling of `Psych::SyntaxError` and not `YAML::ParserError`.

**Comment by treydock** [comment]
Had to also fix up `scripts/functional_gem_behavior.sh` and remove a debugging `puts` that was causing problems.

**Comment by ghoneycutt** [comment]
`lib/rzo/app/config_validation.rb` references `~/.rizzo.json`

**Comment by treydock** [comment]
Updated those references to be ~/.rizzo.yaml.

**Comment by ghoneycutt** [comment]
@treydock Travis is failing. We don't need the pretty check since that does not apply to YAML.

**Comment by treydock** [comment]
Looks like json-schema 2.8.1 broke things.  I forced 2.8.0 and all tests pass.  Looks like the schema URL changed between 2.8.0 and 2.8.1.  I'm updating gemspec to require >=2.8.1.

File: README.md [hunk 1]
```diff
@@ -207,71 +199,54 @@ local, which is the path on the host and owner and group which are the
...
+    forwarded_ports:
+    - guest: "5985"
+      host: "5985"
+    ip: "172.16.100.8"
+    windows: true
+    gui: true
+    shutdown: false
+    update_packages: false
+    box: "mwrock/Windows2016"
+    box_url: ""
```
Comment (ghoneycutt) [inline README.md]: instead of `""` would we use `null` or `~`

Comment (treydock) [reply README.md]: Ah ya, I missed that in README.

**Comment by treydock** [comment]
Updated README to use `~` instead of empty string for values to omit

**Comment by ghoneycutt** [comment]
@treydock the tests are still failing, could you please have a look.

**Comment by treydock** [comment]
@ghoneycutt Fixed, there were a lot of nil values in expected Vagrantfile.  Updated functional test because it seems lack of box was causing nil for box which is actually still allowed.  Should the validation require a box be defined or is no box defined a valid state?  What about memory, should it be omitted if not defined?

This change:

```diff
--- a/scripts/functional_gem_behavior.sh
+++ b/scripts/functional_gem_behavior.sh
@@ -197,9 +197,7 @@ desc "With a valid ~/.rizzo.yaml file looking like:"
 cat > ~/.rizzo.yaml <<EOCONFIG
 ---
   defaults:
-    box: el6-rc5
     bootstrap_repo_path: "${HOME}/git/bootstrap"
-    memory: '1024'
   control_repos:
   - ${PUPPETDATA}
   - ${HOME}/git/ghoneycutt-modules
```

Results in this failure which concerns me:

```diff
rzo generate is expected to produce a Vagrantfile with one VM defined
=====================================================================
--- /tmp/tmp.b8i0rvndEK/vYCQPC.vagrantfile.expected	2018-11-02 17:56:34.024990650 -0400
+++ /tmp/tmp.b8i0rvndEK/hPDOf2.vagrantfile.actual	2018-11-02 17:56:34.436990551 -0400
@@ -8,7 +8,7 @@

   config.vm.define "puppetca", autostart: false do |cfg|
-    cfg.vm.box = "el6-rc5"
+    cfg.vm.box = nil
     cfg.vm.provider :virtualbox do |vb|
-      vb.customize ['modifyvm', :id, '--memory', "1024"]
+      vb.customize ['modifyvm', :id, '--memory', nil]
     end
     cfg.vm.hostname = nil
FAIL: actual Vagrantfile does not match expected file
```

**Comment by treydock** [comment]
I'm gonna patch memory but still curious if the absence of `cfg.vm.box` is valid.

**Comment by ghoneycutt** [comment]
There has to be a box defined at some level for this to be useful. I believe if you don't specify memory, then vagrant uses the box's default.

File: scripts/functional_gem_behavior.sh [hunk 2]
```diff
@@ -199,7 +199,6 @@ cat > ~/.rizzo.yaml <<EOCONFIG
   defaults:
     box: el6-rc5
     bootstrap_repo_path: "${HOME}/git/bootstrap"
-    memory: '1024'
```
Comment (ghoneycutt) [inline scripts/functional_gem_behavior.sh]: why is this removed from the test?

Comment (treydock) [reply scripts/functional_gem_behavior.sh]: The tests before this PR didn't define memory and the memory value was getting defined as nil so left that behavior but ensured no nil values in expected output.

**Comment by treydock** [comment]
@ghoneycutt How should validation be done for box being defined?  Right now no box in default and no box for node will result in `nil` box in Vagrantfile.  The current JSON schema validation would have to be expanded to validate the data after defaults are merged, if I recall correctly.

**Comment by ghoneycutt** [comment]
Let's do that, though in a different PR. Could you please raise an issue for it.

**Comment by ghoneycutt** [comment]
Released in v0.8.0
